libs/dotnet-sdk/FeatureboardInMemoryState.cs

```diff
@@ -0,0 +1,38 @@
+using FeatureboardSdks.DotnetSdk.Models;
+using Microsoft.Extensions.Caching.Memory;
+
+namespace FeatureboardSdks.DotnetSdk;
+
+public class FeatureBoardInMemoryState : IFeatureboardState
+{
+  public DateTimeOffset? LastUpdated { get; private set; }
+  public string? LastModified { get; private set; }
+
+  private readonly IMemoryCache _memoryCache;
+  private const string CacheKey = "FeatureBoardInMemoryState_";
+
+  public FeatureBoardInMemoryState(IMemoryCache memoryCache)
+  {
+    _memoryCache = memoryCache;
+  }
+
+  public Task<FeatureConfiguration?> Get(string featureKey)
+  {
+    return _memoryCache.TryGetValue(CacheKey + featureKey.ToLower(), out FeatureConfiguration? feature)
+      ? Task.FromResult(feature)
+      : Task.FromResult((FeatureConfiguration?)null);
+  }
+
+  public Task AddRange(Dictionary<string, FeatureConfiguration> features, string? lastModified)
```

Comment:
Naming wise, maybe `SetRange` may be better named?

Reply:
AddRange is the name given to adding a set to an existing list or dictionary I wanted to mirror the existing language for this.

Reply:
Code removed / refactored